### DISCUSSION TITLE
Planet rise and set times

### DISCUSSION THREAD

**Discussion by mayankkalbhor** [cosinekitty/astronomy#244] [General]
Does anyone know how to calculate planet rise and set times in javascript with this library

**Comment by cosinekitty** [comment]
Hi @mayankkalbhor and welcome to the project!

To find rise and set times for a planet, use the [SearchRiseSet](https://github.com/cosinekitty/astronomy/tree/master/source/js#SearchRiseSet) function. The Node.js demo program [riseset.js](https://github.com/cosinekitty/astronomy/blob/master/demo/nodejs/riseset.js) shows how to find rise/set times for the Sun and Moon. The same approach works for planets also.

For example, finding the next time Jupiter rises:

```javascript
const riseTime = Astronomy.SearchRiseSet(Astronomy.Body.Jupiter,  observer, +1, date, 300);
```

**Reply by cosinekitty** [reply]
I'm not familiar with that library, but I don't see why both wouldn't work together, because they are separate modules.

**Comment by mayankkalbhor** [comment]
it gives this error: We resolved 'astronomy-bundle' to astronomy-bundle/index.js, but the file does not exist on disk.

please check the codepen and open  codepen console

https://codepen.io/mayanks20/pen/zYjZVre?editors=0011

**Reply by cosinekitty** [reply]
Sorry, I don't know anything about astronomy-bundle. This page is for supporting Astronomy Engine.

**Comment by mayankkalbhor** [comment]
Sorry i mixed 2 things. I am getting the error "Uncaught TypeError: Cannot read properties of null (reading 'SearchRiseSet')" when checking next time Jupiter rises.

please check the codepen and open codepen console

https://codepen.io/mayanks20/pen/MWGmeEP?editors=0011&layout=left

Is there any other import we need to make?

**Reply by cosinekitty** [reply]
I'm not sure how to use skypack... but I tried this:

```javascript
import astronomyEngine from 'https://cdn.skypack.dev/astronomy-engine';
console.log(astronomyEngine);
```

And the result was: `null`.  So for whatever reason, the import isn't working.

**Comment by mayankkalbhor** [comment]
Thanks @cosinekitty  I was able to resolve this. I included browser minified version of astronomy into the script tag and loaded it. Now i am able to fetch the values. 
       const riseTime = Astronomy.SearchRiseSet(Astronomy.Body.Saturn,  observer, +1, new Date(), 300);
        console.log("Saturn Rise Time: ", riseTime.date.toString());
        }

//Saturn Rise Time:  Sun Sep 18 2022 17:46:22 GMT-0500 (Central Daylight Time)
